package/kedro_viz/launchers/cli.py

```diff
@@ -129,6 +131,17 @@ def run(
     """Launch local Kedro Viz instance"""
     from kedro_viz.server import run_server
 
+    kedro_project_path = _find_kedro_project(Path.cwd())
```

Comment:
Why can't we use hook to get this info instead? I feel like the discussion is going into circle.

Reply:
We discussed previously this information is available for all Kedro plugin as long as the cli is extend through the class `KedroCLI`

- https://miro.com/app/board/uXjVKeBIyWU=/
- https://docs.kedro.org/en/stable/api/kedro.framework.cli.hooks.specs.CLICommandSpecs.html#kedro.framework.cli.hooks.specs.CLICommandSpecs

Sure hooks are disable now, but it would be possible to change it to keep only kedro-viz hook.

If you check how `kedro-airflow` works, the `project_metadata` is available (not sure if it's via hook, you may wan to check)
- https://github.com/kedro-org/kedro-plugins/blob/main/kedro-airflow/kedro_airflow/plugin.py

The argument is available since the beginning of the plugin, maybe it's some information just pass around in `click`.

Reply:
i think the idea was also leveraging these two functions so Kedro-viz can run from anywhere inside the kedro directory, jus the way kedro works. I know kedro-mlflow has taken a similar approach. https://github.com/Galileo-Galilei/kedro-mlflow/pull/539

Reply:
https://github.com/kedro-org/kedro/pull/3683 this is exactly why Kedro implements it so that plugin do not have to worry about this. It's fine if kedro-viz want to re-implement it.

Reply:
The issue happens as we run a separate process. Either we need to catch the error before we spawn a process or have some exception queue to communicate the exception to the main process. We opted to catch the error even before we spawn a process. To do so, we can use the implementation here https://github.com/kedro-org/kedro/pull/3683 as you mentioned or duplicate the same on viz.

I am also not a big fan of duplicating logic but since the util function is made internal/private to kedro and we want to reduce coupling with Kedro, we decided to duplicate the implementation on Kedro-Viz. Thank you